Attribute Analysis.ipynb

```diff
@@ -602,6 +606,336 @@
+    "        r1 = np.arange(len(datasets.GROUPINGS_TO_NAMES)-1)\n",
+    "        r1 = [r1[i]+i*4*barWidth for i in range(len(r1))]\n",
+    "        r = [r1]\n",
+    "        for i in range(1, num_attrs+1):\n",
+    "            r.append([x + barWidth for x in r[len(r)-1]])\n",
+    "\n",
+    "        category_attr_ratios = [0 for i in range(len(supercategory_counts[0]))]\n",
+    "        max_categories = [0 for i in range(len(supercategory_counts[0]))]\n",
+    "        for category in range(len(supercategory_counts[0])):\n",
+    "            max_category = 0\n",
+    "            min_category = 1\n",
+    "            max_attr = -1\n",
+    "            for attr in range(num_attrs):\n",
+    "                if supercategory_counts[attr][category] > max_category:\n",
+    "                    max_category = supercategory_counts[attr][category]\n",
+    "                    max_attr = attr\n",
+    "                if supercategory_counts[attr][category] < min_category:\n",
+    "                    min_category = supercategory_counts[attr][category]\n",
+    "            category_attr_ratios[category] = max_category / min_category\n",
+    "            max_categories[category] = max_attr\n",
+    "            \n",
+    "        order = np.argsort(category_attr_ratios)\n",
+    "        r = np.array(r)\n",
+    "        for i in range(num_attrs-1, -1, -1):\n",
+    "            plt.barh(r[i], supercategory_counts[i][order], height=barWidth, color=COLORS[i], edgecolor='white', label=attr_names[i])\n",
+    "        \n",
+    "        ticks = [r[0]+(num_attrs/2)*barWidth for i in range(len(r1))][0]\n",
+    "        plt.yticks(ticks, np.array([datasets.GROUPINGS_TO_NAMES[i+1] for i in range(len(r1))])[order], fontsize=fontsize)\n",
+    "        plt.xticks(fontsize=fontsize)\n",
+    "        plt.legend(loc='best', fontsize=fontsize)\n",
+    "        plt.ylabel('Object Category', fontsize=fontsize, labelpad=20)\n",
+    "        plt.xlabel('Fraction of Labelled Images that contain this Category', fontsize=fontsize, labelpad=20, x=.3)\n",
+    "        plt.tight_layout()\n",
+    "        plt.gcf().subplots_adjust(bottom=0.18)\n",
+    "        plt.gcf().subplots_adjust(left=0.28)\n",
+    "        if first_pass:\n",
+    "            to_write[3] = ['(M2) Distribution of object categories each attribute appears with, sorted by ratio between the analyzed attribute and everything else.']\n",
+    "            plt.savefig(\"results/{0}/{1}/1.png\".format(folder_name, save_loc))\n",
+    "        plt.show()\n",
+    "\n",
+    "# Graphs the ratio of instance counts if they are statistically significant\n",
+    "indices_to_keep = [i for i in range(len(instance_counts[0])) if categories[i] not in dataset.people_labels]\n",
+    "instance_sig = [norm_instance_counts[i][indices_to_keep] for i in range(num_attrs)]\n",
+    "xaxis_sig = np.array(xaxis)[indices_to_keep]\n",
+    "pvalues_sig = np.array(p_values)[indices_to_keep]\n",
```

Comment:
I don't think this line makes any sense, given the logic above that calculates what p_values is